pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -108,82 +108,62 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 
 	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
-		logger.V(1).Info("The VM is not being deleted")
-		// The object is not being deleted, so if it does not have a finalizer,
-		// then lets add the finalizer and update the object.
-		if !helper.ContainsString(instance.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName) {
-			err = r.addFinalizerAndUpdate(&request, logger)
+		logger.V(1).Info("vm create/update event")
+		// The object is not being deleted, so we can set the macs to allocated
+		err = r.poolManager.MarkVMAsReady(instance, logger)
```

Comment:
question: MarkVMAsReady is to be done after VM is created right ?

Reply:
yes. **successfully** created...